src/components/manifold-subscription-details/manifold-subscription-details.tsx

```diff
@@ -17,18 +17,23 @@ export class ManifoldSubscriptionDetails {
   @Prop() subscriptionId: string;
   @Prop() heading?: string;
   @Prop() isEditing?: boolean = false;
+  @Prop() preview?: boolean = false;
 
   // Can this be abstracted/optimized further?
   async componentWillLoad() {
-    setConnection(await getManifoldConnection(this.el));
+    if (!this.preview) {
+      setConnection(await getManifoldConnection(this.el));
+    }
```

Comment:
The subscription create component will likely need a connection in preview mode to fetch _some_ real data, while mocking others. Here, too, I think the goal was for a provider to be able to see their plan data, but without actually charging a subscription.

Maybe we always set the connection no matter what, but then—how would we short-circuit skipping loading some data in preview mode? Maybe in the method for each query?

Reply:
> how would we short-circuit skipping loading some data in preview mode?

How would we skip loading data if we want to show the actual plans? wouldn't loading it be necessary?

Reply:
Oh sorry. I was trying to say we load the actual data for the plans, but skip loading everything else, if the. I haven’t looked closely at this component, and wasn’t sure if there was any data that needed to be loaded for subscriptions or auth.

Reply:
Also let me know if loading plan data w/o auth is tricky! We can talk through it more if so.

Reply:
hmm, this might be tricky. this component takes in a subscription ID and gets the associated plan based on that. subscriptions need auth, otherwise we would have to pass in a plan ID and have a completely different set of queries for the preview state. at that point it's almost a completely separate component

Reply:
might have some ideas of how this could work. I'll give it a try. let me know if you have any ideas though